spec.mediawiki

```diff
@@ -310,32 +291,24 @@ where '''''order<sub>i</sub>''''' represents the order ID of the ith order in a
 lexicographically sorted list of order IDs, '''''||''''' indicates 
 concatenation, and '''''/''''' is integer division.
 
-Each order's unique ID and corresponding hash are assigned as the order is 
-placed, and published at the start of the matching process.
+The order IDs and final hash are published at the start of the matching process.
 This is done so that the shuffling may be independently computed for external 
 verification and to facilitate server mesh operation.
-Section 5 describes the order submission and ID assignment.
 
-Cancellations are processed first. Cancellation can potentially carry a penalty 
-if the client's cancel:complete ratio falls below community conduct guidelines. 
+Orders are processed one at a time. Each order is matched according to its type.
 
-Having prioritized each taker order in the epoch pool with a pseudorandom and 
-verifiably unbiased process, matching begins with the first order in the list.
-The order is paired with the best possible 
-(in terms of rate) compatible standing limit order(s), and the orders are queued 
-for monitoring by the DEX. Both clients are immediately informed.
+1. If this current order is a '''cancel order''', any corresponding standing limit order is removed from the list and the cancel order is considered filled. If a cancellation is processed before the order that it cancels, the cancellation will fail, and will need to be re-submitted.
 
-The process continues with the next order in the list and iterates until all 
-orders are removed from the list.
+2. If the order is a limit order than cannot immediately match ('''a maker'''), it is added to the standing orders. It is immediately able to match orders further down in the queue.
+
+3. If the order is a '''taker''', it is matched against the best available standing order. Clients for both orders are notified and the settlement process begins. The orders are set aside for monitoring. If a limit order with time in force "standing" on either side of the match is only partially filled, it is added to the standing orders with the appropriate modifications and is immediately available for matching again. 
 
-New limit orders are then added to the order book. 
-Any taker limit orders which are not type "Immediate" and which did not fully 
-match are added to the order book as well.
+In the special case of a standing limit order in the queue matching immediately, 
+the price assigned to the match is the average of the two orders, with a split 
+atom going to the maker.
```

Comment:
Not  sure about this part.

Reply:
I'm not totally certain which cases are covered by "a standing limit order in the queue matching immediately", but what occurred to me previously are two limit orders that match immediately (i.e. after the first is added to the books and the second is matched with the first).  On some exchanges I see Mid Market Price shown on the order book.  Applying that is fair to both maker and taker, and I don't see an issue with that because in this scenario both limit orders are getting prices that respect their limits (nothing to complain about as long as it's fair).